File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        
        int l = 0;
        int r = height.length - 1;
        int max_ = 0;
        while(l < r){
            
            int h = Math.min(height[l], height[r]);
            int area = h * (r - l);
            max_ = Math.max(max_, area);
            
            if(height[l] < height[r]){
                l++;
            } else {
                r--;
            }
        }
        return max_;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        l = 0 
        r = len(height) - 1
        
        max_ = 0
        while(l < r):
            
            h = min(height[l], height[r])
            area = h * (r-l)
            max_ = max(max_, area)
            
            if(height[l] < height[r]):
                l+=1
            else:
                r-=1
        
        return max_
        